addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/alerts/WebSocketAlertWrapper.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    public WebSocketMessageDTO getWebSocketMessageDTO() {
+        return webSocketMessageDTO;
+    }
+
+    public String getSolution() {
+        return alert.getSolution();
+    }
+
+    public String getReference() {
+        return alert.getReference();
+    }
+
+    public String getEvidence() {
+        return alert.getReference();
+    }
+
+    public int getCweIdm() {
+        return alert.getCweId();
+    }
+
+    public int getWascId() {
+        return alert.getWascId();
+    }
+
+    public HttpMessage getHandshakeMessage() {
+        return alert.getMessage();
+    }
+
+    public Alert.Source getSource() {
+        return alert.getSource();
+    }
+
+    public HistoryReference getHandshakeReference() {
+        return alert.getHistoryRef();
+    }
+
+    public int getAlertId() {
+        return alert.getAlertId();
+    }
+
+    public int getConfidence() {
+        return alert.getConfidence();
+    }
+
+    public int getRisk() {
+        return alert.getRisk();
+    }
+
+    public String getName() {
+        return alert.getName();
+    }
+
+    public String getMessageReadablePayload() {
+        try {
+            return webSocketMessageDTO.getReadablePayload();
+        } catch (InvalidUtf8Exception e) {
+            return Constant.messages.getString("websocket.payload.invalid_utf8");
+        }
+    }
+
+    public abstract static class WebSocketAlertBuilder {
+
+        private Alert alert;
+
+        private static final String OTHER_INFO_LABEL = "[WebSocket Message] ";
+
+        private WebSocketMessageDTO webSocketMessageDTO;
+
+        public WebSocketAlertBuilder(int pluginId, Alert.Source source) {
+            alert = new Alert(pluginId);
+            alert.setSource(source);
+        }
+
+        public WebSocketAlertBuilder setName(String name) {
+            alert.setName(name);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setDescription(String description) {
+            alert.setDescription(description);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setParam(String param) {
+            alert.setParam(param);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setAttack(String attack) {
+            alert.setAttack(attack);
+            return this;
+        }
+
+        protected WebSocketAlertBuilder setMessage(WebSocketMessageDTO webSocketMessageDTO) {
+
+            HttpMessage handshakeMessage;
+
+            try {
+                handshakeMessage =
+                        webSocketMessageDTO.channel.getHandshakeReference().getHttpMessage();
+            } catch (Exception e) {
+                LOGGER.info(
+                        "Couldn't get the Handshake Http Message for this specific channel. "
+                                + "Channel ID:"
+                                + webSocketMessageDTO.channel.id,
+                        e);
+                return null;
```

Comment:
Should return `this` still? Otherwise it will lead to a NPE.

Reply:
Fixed in 976c31e